src/lightcurvelynx/math_nodes/ra_dec_sampler.py

```diff
@@ -72,42 +74,100 @@ def compute(self, graph_state, rng_info=None, **kwargs):
+        Default: False
+    **kwargs : dict, optional
+        Additional keyword arguments to pass to the parent class constructor.
     """
 
-    def __init__(self, data, radius=None, in_order=False, **kwargs):
-        if radius is None:
+    def __init__(self, data, *, extra_cols=None, radius=None, in_order=False, **kwargs):
+        if radius is None and isinstance(data, ObsTable):
             radius = data.survey_values.get("radius", None)
-            if radius is None:
-                raise ValueError("ObsTable has no radius. Must provide radius.")
-        if radius < 0.0:
-            raise ValueError("Invalid radius: {radius}")
+        if radius is None or radius < 0.0:
+            raise ValueError(f"Invalid radius: {radius}")
         self.radius = radius
 
+        # Start with RA, dec, and (optionally) time.
         data_dict = {
             "ra": data["ra"],
             "dec": data["dec"],
-            "time": data["time"],
         }
+        if "time" in data:
+            data_dict["time"] = data["time"]
+
+        # Add any extra columns (without duplicates).
+        if extra_cols is not None:
+            for col in extra_cols:
+                if col not in data_dict:
+                    data_dict[col] = data[col]
+
         super().__init__(data_dict, in_order=in_order, **kwargs)
 
+    @classmethod
+    def from_hats(cls, path, *, radius=None, extra_cols=None, in_order=False, **kwargs):
+        """Create a GivenRADECSampler from the observations in a HATS Catalog.
+
+        Parameters
+        ----------
+        path : str or Path
+            The base path of the HATS data directory.
+        radius : float, optional
+            The sampling radius (noise factor) in degrees. Use 0.0 to return
+            the exact pointings.
+            Default: 0.0
+        extra_cols : list of str, optional
+            A list of extra column names to include in the sampling.
+            Default: None
+        in_order : bool
+            Return the given data in order of the rows (True). If False, performs
+            random sampling with replacement.
+            Default: False
+        **kwargs : dict, optional
+            Additional keyword arguments to pass to the constructor.
+
+        Returns
+        -------
+        GivenRADECSampler
+            The created GivenRADECSampler object.
+        """
+        # See if the (optional) LSDB package is installed.
+        try:
+            from lsdb import read_hats
+        except ImportError as err:
+            raise ImportError(
+                "The lsdb package is required to read HATS catalogs. "
+                "Please install it via 'pip install lsdb'."
+            ) from err
+
+        # Compute the full list of columns to load from HATS.
+        cols_to_load = ["ra", "dec"]
+        if extra_cols is not None:
+            cols_to_load.extend(extra_cols)
+        columns = list(set(cols_to_load))  # Remove any duplicates.
```

Comment:
I don't know if it matters for the rest of the pipeline, but the order of columns could differ from time to time.

Reply:
It shouldn't impact the rest of the pipeline. I was thinking about a few different ways of trying to get around the different names, but don't have a great suggestion beyond people including it in the extra columns.